File: Day1/SecondLargest.py
#Link URL:- https://www.geeksforgeeks.org/batch/gfg-160-problems/track/arrays-gfg-160/problem/move-all-zeroes-to-end-of-array0751


def secondLargest(nums):
    n=len(nums)

    secLar=-1
    lar = max(nums)
    for i in range(n):
        if(nums[i]<lar):
            secLar=max(secLar,nums[i])
    return secLar





t = int(input())
nums = list(map(int,input().split()))
print(f"Second Largest Number:{secondLargest(nums)}")

File: Day1/SecondLargest.cpp
// Link URL:- https://www.geeksforgeeks.org/batch/gfg-160-problems/track/arrays-gfg-160/problem/move-all-zeroes-to-end-of-array0751


#include <bits/stdc++.h>
using namespace std;

int secondLargest(vector<int> &nums){

    int n = nums.size();
    int max_ele = *max_element(nums.begin(),nums.end());

    int sec_max=-1;
    for(int i=0;i<n;i++){
        if(nums[i]<max_ele){
            sec_max=max(nums[i],sec_max);
        }
    }

    return sec_max;

}

int main(){

    int t;
    cin>>t;
    vector<int> nums;
    while(t--){
        int n;
        cin>>n;
        nums.push_back(n);

    }

    cout<<secondLargest(nums)<<endl;

};
